### PR TITLE
Move contact form static assets into packages/forms

### PR DESCRIPTION
**Pull request by ice9js**
Part of #28405.

## Proposed changes:

This patch moves the contact form static files from `plugins/jetpack/modules/contact-form` into `packages/forms/src/contact-form`.  
These files are being moved as is, with no changes whatsoever, which is why I decided to split it up from the PHP part which will contain changes.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pcsBup-N3-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

This patch only copies some files and has no impact on any functionality.

### CODE REVIEW

**Review by jeherve — CHANGES_REQUESTED** [review]
It looks like we'll need to bring in our custom linting rules for those old files from `projects/plugins/jetpack/modules/.eslintrc.js`, or we will not be able to merge. 

We shouldn't assign those linting rules to all the package, but only to the old files.

**Review by jeherve — COMMENTED** [review]
Linting seems to be okay now, nice work.

Maybe we can remove the files that were .gitignored in the Jetpack plugin (.map, .min, and .rtl files)?

**Comment by ice9js** [comment]
>Maybe we can remove the files that were .gitignored in the Jetpack plugin (.map, .min, and .rtl files)?

Makes total sense. I guess this also means we'll need to set things up to build those... I'll start looking into this.

**Comment by jeherve** [comment]
> I guess this also means we'll need to set things up to build those... I'll start looking into this.

If that can help: we have used webpack for those in the Jetpack plugin until now. The configuration is available here:
`projects/plugins/jetpack/tools/webpack.config.css.js`

We should be able to start using webpack in the new package as well, in a similar way.

**Comment by ice9js** [comment]
@jeherve I added a `webpack.config.contact-form.js` for building the CSS files taking inspiration from the file you mentioned and it seems to work when I run it with `pnpm --filter @automattic/jetpack-forms build`. Just two caveats remaining:

- It also produces a `jetpack-rtl.css`, `jetpack.css` and `jetpack.css.map` which weren't there before so I'm not sure if we need them. But I'm also not sure what's adding them 🤔.
- Is the `build` command automatically picked up when you run `jetpack install packages/forms` or do we still need to hook it up somewhere?

**Comment by jeherve** [comment]
> * It also produces a `jetpack-rtl.css`, `jetpack.css` and `jetpack.css.map` which weren't there before so I'm not sure if we need them. But I'm also not sure what's adding them 🤔.

I don't think those will be necessary. I think that's also our opportunity to simplify that whole webpack config file. We do not need to bother with renames here for example, since we'll have the opportunity to update the way all those files are enqueued when we bring the PHP over.

> * Is the `build` command automatically picked up when you run `jetpack install packages/forms` or do we still need to hook it up somewhere?

If you update `composer.json`, you'll be able to run `jetpack build packages/search` afterwards. Like so:

```diff
diff --git a/projects/packages/forms/composer.json b/projects/packages/forms/composer.json
index ae97eea244..23ce0e73ee 100644
--- a/projects/packages/forms/composer.json
+++ b/projects/packages/forms/composer.json
@@ -24,8 +24,12 @@
 		"test-php": [
 			"@composer phpunit"
 		],
-		"build-production": "echo 'Add your build step to composer.json, please!'",
-		"build-development": "echo 'Add your build step to composer.json, please!'",
+		"build-production": [
+			"pnpm run build-production"
+		],
+		"build-development": [
+			"pnpm run build"
+		],
 		"post-install-cmd": "WorDBless\\Composer\\InstallDropin::copy",
 		"post-update-cmd": "WorDBless\\Composer\\InstallDropin::copy"
 	},
diff --git a/projects/packages/forms/package.json b/projects/packages/forms/package.json
index 1352fb3cf3..b0546deebc 100644
--- a/projects/packages/forms/package.json
+++ b/projects/packages/forms/package.json
@@ -16,6 +16,7 @@
 	"author": "Automattic",
 	"scripts": {
 		"build": "pnpm run build:contact-form",
+		"build-production": "NODE_ENV=production BABEL_ENV=production pnpm run build",
 		"build:contact-form": "NODE_PATH=\"$PWD/node_modules\" webpack --config ./tools/webpack.config.contact-form.js",
 		"clean": "true"
 	},

```

**Comment by ice9js** [comment]
I somewhat simplified the webpack config and added the recommended changes to `composer.json`.  
I'm sure the config could use some more work, however I'd prefer to do that separately so we can shift to working in the new package sooner and avoid synchronization issues down the line.

File: projects/packages/forms/changelog/update-move-contact-form-static-assets [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
```
Comment (CGastrell) [inline projects/packages/forms/changelog/update-move-contact-form-static-assets:1]: I don't think "minor" is intended for this kind of things :D

Comment (ice9js) [reply projects/packages/forms/changelog/update-move-contact-form-static-assets:1]: It's not breaking compatibility, or doing much really. Just adding some files 🙂

**Review by CGastrell — DISMISSED** [review]
IPF! :shipit:

**Review by jeherve — APPROVED** [review]

**Pull request merged**